cin_validator/rules/cin2022_23/rule_8545Q.py

```diff
@@ -0,0 +1,106 @@
+from typing import Mapping
+
+import pandas as pd
+
+from cin_validator.rule_engine import (
+    CINTable,
+    IssueLocator,
+    RuleContext,
+    RuleType,
+    rule_definition,
+)
+from cin_validator.test_engine import run_rule
+from cin_validator.utils import make_census_period
+
+ChildIdentifiers = CINTable.ChildIdentifiers
+Header = CINTable.Header
+PersonDeathDate = ChildIdentifiers.PersonDeathDate
+Year = Header.Year
+
+# define characteristics of rule
+@rule_definition(
+    code="8545Q",
+    module=CINTable.ChildIdentifiers,
+    rule_type=RuleType.QUERY,
+    message="Please check: Child's date of death should be within the census year",
+    affected_fields=[PersonDeathDate],
+)
+def validate(
+    data_container: Mapping[CINTable, pd.DataFrame], rule_context: RuleContext
+):
+    df = data_container[ChildIdentifiers]
+    df_ref = data_container[Header]
+
+    """
+    If present, <PersonDeathDate> (N00108) must be within [Period_of_Census]
+    """
+
+    df = df[[PersonDeathDate]]
+
+    # Death date must not be null, invalid text dates are made null in the line above
+    df = df[df[PersonDeathDate].notna()]
+
+    # Create a key column to join
+    # df["Key"] = 0
+    """df_ref = df_ref[[ReferenceDate]]
+    df_ref["YearEnd"] = pd.to_datetime(df_ref["ReferenceDate"], format=r"%Y-%m-%d", errors="coerce") #equals 31/03/2023  (for example, year can be different)
+    df_ref["YearStart"] = df_ref["YearEnd"] - pd.DateOffset(years=1) + pd.DateOffset(days=1)         #equals 01/04/2022  (for example, year can be different)
+    #Create a key column to join
+    df_ref["Key"] = 0"""
+    # df = df.merge(df_ref, on="Key")
+
+    collection_year = df_ref[Year]
```

Comment:
The way you used `ReferenceDate` here was correct also.
I changed it to use Year because that's what the `make_census_period` function expects.